.github/workflows/cargo-update.yml

```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout the repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
```

Comment:
Why comment the version behind the hash? Aren't versions ultimately meaningless when we don't use them anyway?

Will the next dependabot update not just break this, causing pointless additional work having to update the version every time?

Something like @v4 can get updated, but can a tag like @v4.1.0 get updated? Doesn't that always link to one specific version anyway? Or is it fully up to the action creator on which version they want to overwrite.

Reply:
Dependabot will update these version comments at the same time that it updates the pin ([Reference](https://github.com/dependabot/dependabot-core/issues/4691#issuecomment-1296532496)). There is no manual update required for either the hash or the version comment. This is nice because it makes it easier to map the pin to the tagged version without having to look it up. The same thing is done [for `pre-commit` hooks in `phylum-ci`](https://github.com/phylum-dev/phylum-ci/blob/main/.pre-commit-config.yaml) and that has worked out well.

Yes...action creators are responsible for bumping tags. The convention most follow is to provide specific/fixed patch versions (e.g., vX.Y.Z) and floating major and minor versions (e.g., vX and vX.Y). See our own [phylum-dev/phylum-analyze-pr-action tags](https://github.com/phylum-dev/phylum-analyze-pr-action/tags) for an example of this. Currently, these three tags all point to the same hash: v2.1.1, v2.1, and v2
When v2.1.1 was released, the tags for v2.1 and v2 were overwritten to point away from the v2.1.0 release.

Reply:
But v2.1.0 is still the same, correct? If these are just git tags without any immutability assigned to them then that's quite honestly just very poor design.

Reply:
Yes...v2.1.0 is still the same...but only by convention. Git tags can be updated at any time to point to new/different commit SHAs. This PR is meant to get away from the risk of that design where a tag can shift out from underneath us. The Dependabot method of tracking the version along with the tag's commit is just an added convenience.